var/spack/repos/builtin/packages/migraphx/package.py

```diff
@@ -38,17 +39,21 @@ def url_for_version(self, version):
     depends_on('protobuf', type='link')
     depends_on('blaze', type='build')
     depends_on('nlohmann-json', type='link')
-    depends_on('py-pybind11', type='build')
     depends_on('msgpack-c', type='link')
     depends_on('half@1.12.0', type='link')
 
-    for ver in ['3.5.0', '3.7.0', '3.8.0', '3.9.0', '3.10.0', '4.0.0']:
+    for ver in ['3.5.0', '3.7.0', '3.8.0', '3.9.0', '3.10.0', '4.0.0', '4.1.0']:
         depends_on('hip@' + ver, type='build', when='@' + ver)
         depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
         depends_on('llvm-amdgpu@' + ver, type='build', when='@' + ver)
         depends_on('rocblas@' + ver, type='link', when='@' + ver)
         depends_on('miopen-hip@' + ver, type='link', when='@' + ver)
-        if ver in ['3.9.0', '3.10.0', '4.0.0']:
+        if ver in ['4.1.0']:
+            depends_on('py-pybind11@2.6.1', type='build')
+        else:
+            depends_on('py-pybind11', type='build')
```

Comment:
```suggestion
    depends_on('py-pybind11', type='build')
    depends_on('py-pybind11@2.6.1', type='build', when='@4.1.0')
```

I would be in favor of dropping all this control flow `for` `if` etc and try to make it declarative when possible, since that's easier to follow.